api/swagger.yaml

```diff
@@ -0,0 +1,282 @@
+servers:
+  - url: 'https://chat-dancier.net'
+  - url: 'http://localhost:8080'
+paths:
+  /chats:
+    get:
+      parameters:
+        - $ref: '#/components/parameters/user_id'
+      summary: returns a list of all chats of a user. The list endpoint does only return the latest message, the messages array will only be filled on fetching messages for a chat
+      responses:
+        '200':
+          description: "list of chats for a user"
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/chat_list_response'
+        default:
+          description: Unexpected error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/error'
+    post:
+      summary: Creates a new chat group
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/chat_creation'
+      responses:
+        '201':
+          description: "chat created"
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/chat_response'
+        default:
+          description: Unexpected error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/error'
+  /chats/{chat_id}:
+    get:
+      summary: returns a chat
+      parameters:
+        - $ref: '#/components/parameters/chat_id'
+      responses:
+        '200':
+          description: "data of a chat"
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/detailed_chat'
+        default:
+          description: Unexpected error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/error'
+  /chats/{chat_id}/messages:
+    get:
+      summary: returns messages for a chat
+      parameters:
+        - $ref: '#/components/parameters/chat_id'
+        - $ref: '#/components/parameters/last_message_id'
+      responses:
+        '200':
+          description: "detailed data of a chat"
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/message_list_response'
+        default:
+          description: Unexpected error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/error'
+    post:
+      summary: Creates a new message
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/base_message'
+      responses:
+        '201':
+          description: "message created"
+        default:
+          description: Unexpected error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/error'
+  /new-messages-long-polling:
```

Comment:
I have to invest some time to understand this. Will have a look on this this evening (I hope)